### PR TITLE
WIP: Auth

### PR DESCRIPTION
**Pull request by p4r4n01d**
This is ok to merge, but there are lots of conflicts needing resolution. This adds support for accessing protected Gerrits, starring changes, UI tweaks to the commit cards and numerous bug fixes. I once we do a release, this PR can be closed and a new one can be opened for the remaining items on the list.

#### Progress and timeline breakdown
- [x] Sign in layout and activity
- [x] Authenticate with user's username and password and get account details
- [x] Storing and retrieval of user's account details
- [x] Sign in activity updates including button feedback and authentication error messages
- [x] Fetch user's starred changes
- [x] Commit card design tweaks for starring changes
- [x] Ability to star changes
- [x] Clear starred changes when logging out
- [ ] **Possible first release**
- [ ] Left side navigation draw
- [ ] Basic commenting (text only)
- [ ] Advanced Commenting with ratings
- [ ] Add reviewers

### CODE REVIEW

File: src/com/jbirdvegas/mgerrit/objects/AccountEndpoints.java [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+    // --- Parcelable stuff so we can send this object through intents ---
+    public static final Parcelable.Creator<AccountEndpoints> CREATOR
+            = new Parcelable.Creator<AccountEndpoints>() {
+        public AccountEndpoints createFromParcel(Parcel in) {
+            return new AccountEndpoints(in);
+        }
+
+        @Override
+        public AccountEndpoints[] newArray(int size) {
+            return new AccountEndpoints[0];
```
Comment (JBirdVegas) [inline src/com/jbirdvegas/mgerrit/objects/AccountEndpoints.java]: Shouldn't this be
```
return new AccountEndpoints[size];
````

Comment (p4r4n01d) [inline src/com/jbirdvegas/mgerrit/objects/AccountEndpoints.java]: Yeah that would be more correct. However, we never use an array of URLs
(formally GerritURLs) and there is no current reason for working with them.
We have one message per URL and I cannot see a need for this to change.
That's is the reason this bug remained for so long (I copied this from
GerritURL).

**Comment by jstralko** [comment]
Hello, 

Whats the best way to merge this into my local git repo?  I want to test this out.   Should I just do a 

curl -v 'https://github.com/JBirdVegas/external_jbirdvegas_mGerrit/pull/11.diff' -O 
git apply 11.diff

Thanks,

Gerb.

**Comment by p4r4n01d** [comment]
That should work, although as this repo is ahead of the AOKP one, make sure
you merge:
"Save the name of the current Gerrit name in the preferences" (
https://github.com/JBirdVegas/external_jbirdvegas_mGerrit/commit/72815d10625f355085910446007e1d4b02dec0c7
)
As that is required for the title of the sign in screen. This commit
couldn't hurt either:
"Set the main change list title to the name of the current Gerrit" (
https://github.com/JBirdVegas/external_jbirdvegas_mGerrit/commit/b89cace057c8e9eef8e1723fb97be08bec3e72b4
)

Evan.

**Comment by JBirdVegas** [comment]
@jstralko You could also get it by adding this repo as a remote to you current local repo.

Something like this.
git remote add jbirdRepo https://github.com/JBirdVegas/external_jbirdvegas_mGerrit
git fetch jbirdRepo
git checkout jbirdRepo/auth
gradle clean build

**Comment by p4r4n01d** [comment]
@jstralko I have done a lot of work on authentication - adding the ability to login and star/unstar changes. Can you test that this works with a Gerrit instance which requires a username and password. After adding it, you should be prompted for the username and password, and it should authenticate all further requests. There is an issue where the refresh indicator is not coming up, but the changes should appear after a few seconds.

@JBirdVegas This is nearly ready for an initial release. I have one small item left and would like to do some more testing first.

**Comment by JBirdVegas** [comment]
Awesome! I'm finishing up the final stages in the new jenkins release build bot.  If you release before the bot is complete be careful to increment `version.properties`

**Comment by p4r4n01d** [comment]
@JBirdVegas I merged master into this branch and can confirm that it builds/runs fine locally. Can you have a look at why the Travis build is failing? I am not sure how you set that up with the private credentials file (and I don't have this file). After that this is ready to be merged and released. I won't have time until next weekend, so you can release before then if you have time. Let me know and I can provide some text for the what's new and I'll also update the screenshots.

**Comment by p4r4n01d** [comment]
-1: There is a crash when rotating the device. Release will have to wait another week.
Binary XML file line #45: Error inflating class fragment
Caused by: java.lang.IllegalStateException: Fragment com.jbirdvegas.mgerrit.fragments.ChangeListFragment did not create a view.

**Comment by p4r4n01d** [comment]
@jbirdvegas IReleases will fail fast, as long as you don't attempt a release build is fail-safe.

**Comment by JBirdVegas** [comment]
@p4r4n01d, That's my fault I have a fix, I'll push it tonight when I get home.  I'll also email you about how to perform a release.  I'd say your local releases are failing because of the jenkins commits, I'll explain the fix in email. Apologies I ment to email you last night but forgot.

Basically the if condition in the build.gradle that checks if private.creds exists is missing an else. If the file does not exist we must set android.signingConfigs = null.  It was an oversight I noticed this morning

**Comment by JBirdVegas** [comment]
And Yea I'll reword the warning

**Comment by JBirdVegas** [comment]
@p4r4n01d Travis build is fixed now and I reworded the warning to be clearer.  Feel free to merge to master and release :+1:

**Pull request merged**